libsolidity/formal/SMTEncoder.cpp

```diff
@@ -738,6 +742,32 @@ void SMTEncoder::visitGasLeft(FunctionCall const& _funCall)
 		m_context.addAssertion(symbolicVar->currentValue() <= symbolicVar->valueAtIndex(index - 1));
 }
 
+void SMTEncoder::visitAddMod(FunctionCall const& _funCall)
+{
+	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
+	solAssert(funType.kind() == FunctionType::Kind::AddMod, "");
+	auto const& args = _funCall.arguments();
+	solAssert(args.at(0) && args.at(1) && args.at(2), "");
+	auto x = expr(*args.at(0));
+	auto y = expr(*args.at(1));
+	auto k = expr(*args.at(2));
+	m_context.addAssertion(k != 0);
+	defineExpr(_funCall, (x + y) % k);
+}
+
+void SMTEncoder::visitMulMod(FunctionCall const& _funCall)
+{
+	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
```

Comment:
I think this is great example where to use auto. Even clang-tidy complains about it :)
```suggestion
	auto const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
```